File: crowdin/src/main/java/com/crowdin/platform/repository/remote/RemoteRepository.java
package com.crowdin.platform.repository.remote;

import android.content.Context;

import com.crowdin.platform.api.LanguageData;
import com.crowdin.platform.repository.LanguageDataCallback;

/**
 * Repository of strings from network.
 */
public interface RemoteRepository {

    /**
     * Save {@link LanguageData} for a specific language.
     *  @param currentLocale        the default device Locale.
     * @param languageDataCallback delivers data back to caller.
     */
    void fetchData(Context context, String currentLocale, LanguageDataCallback languageDataCallback);
}

File: crowdin/src/main/java/com/crowdin/platform/repository/remote/RemoteRepository.kt
package com.crowdin.platform.repository.remote

import com.crowdin.platform.repository.LanguageDataCallback
import com.crowdin.platform.repository.model.LanguageData

/**
 * Repository of strings from network.
 */
internal interface RemoteRepository {

    /**
     * Save [LanguageData] for a specific language. Executes on a background thread.
     *
     * @param languageDataCallback  delivers data back to caller.
     */
    fun fetchData(languageDataCallback: LanguageDataCallback) {}
}
